File: Source/Example.Chat.Server/Messages.cs
using System;
using System.Linq;

namespace Example
{
    [Serializable]
    public class NotificationMessage
    {
        public string Text { get; set; }
    }

    [Serializable]
    public class NewMessage
    {
        public string Username { get; set; }
        public string Text { get; set; }
    }
}

File: Source/FSharp.Example.Observers.Chat.Shared/Messages.fs
module Messages

open System.Linq
open System.Collections.Generic

open Orleankka
open Orleankka.FSharp

type ClientMessage =
   | NewMessage of Username:string * Text:string
   | Notification of Text:string

type ServerMessage =
   | Join of Username:string * Client:ObserverRef
   | Say of Username:string * Text:string   
   | Disconnect of Username:string * Client:ObserverRef